x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -137,7 +144,23 @@ protected AllocatedPersistentTask createTask(long id, String type, String action
                                                  PersistentTasksCustomMetadata.PersistentTask<ShardFollowTask> taskInProgress,
                                                  Map<String, String> headers) {
         ShardFollowTask params = taskInProgress.getParams();
-        Client followerClient = wrapClient(client, params.getHeaders());
+        final Client followerClient;
+        if (Assertions.ENABLED) {
+            followerClient = new FilterClient(wrapClient(client, params.getHeaders())) {
+                @Override
+                protected <Request extends ActionRequest, Response extends ActionResponse>
+                void doExecute(ActionType<Response> action, Request request, ActionListener<Response> listener) {
+                    if (request instanceof MasterNodeRequest) {
+                        final TimeValue masterTimeout = ((MasterNodeRequest<?>) request).masterNodeTimeout();
+                        assert masterTimeout.nanos() == Long.MAX_VALUE :
```

Comment:
Instead of checking this here, could we add a transport interceptor in the CCR IT tests that asserts the same thing?

Reply:
ok, I've moved this check to tests. See https://github.com/elastic/elasticsearch/pull/60070/commits/de4fc6153915034a1b6cbcfb82b0a205342fed46.